# Github Repository Overview
This is an overview of all github repositories associated with [PuckerLab](https://www.tu-braunschweig.de/en/ifp/pbb). Please find a list of the science communication projects, the developed bioinformatics tools, teaching activities, and project specific repositories below.


## Science Communication
### MybMonday
Specific facts and research projects about MYB transcription factors are shared on a weekly basis via Twitter and LinkedIn using [#MybMonday](https://twitter.com/search?q=%23MybMonday&src=typed_query). All posts are collected in the [#MybMonday github repository](https://github.com/bpucker/MybMonday).

<a href="https://github.com/bpucker/MybMonday">
<img alt="Phylogenetic tree of banana MYBs (Pucker <i>et al.</i>, 2020; doi:10.1371/journal.pone.0239275)" src="https://www.biorxiv.org/content/biorxiv/early/2020/02/03/2020.02.03.932046/F2.large.jpg" width="30%" height="30%">
</a>

### FlavonoidFriday
Specific facts and research projects about the flavonoid biosynthesis are shared on a weekly basis via Twitter and LinkedIn using [#FlavonoidFriday](https://twitter.com/search?q=%23FlavonoidFriday&src=typed_query&f=top). All posts are collected in the [#FlavonoidFriday github repository](https://github.com/bpucker/FlavonoidFriday).

<a href="https://github.com/bpucker/FlavonoidFriday">
<img alt="Flavonoid biosynthesis pathway overview (Pucker <i>et al.,</i> 2020; doi:10.3390/plants9091103" src="https://www.mdpi.com/plants/plants-09-01103/article_deploy/html/images/plants-09-01103-g001.png" width="25%" height="25%">
</a>  


## Tools
### Knowledge-based Identification of Pathway Enzymes (KIPEs)
Knowledge-based Identification of Pathway Enzymes (KIPEs) performs an automatic annotation of the flavonoid biosynthesis steps in a new transcriptome of genome sequence assembly. Scripts and datasets are availabe in the [KIPEs repository](https://github.com/bpucker/KIPEs). KIPEs is also available on [our web server](http://pbb.bot.nat.tu-bs.de/KIPEs/). This enables all scientists to use KIPEs in their research projects. Please find additional details in the corresponding publication ['Automatic Identification of Players in the Flavonoid Biosynthesis with Application on the Biomedicinal Plant <i>Croton tiglium</i>'](https://www.mdpi.com/2223-7747/9/9/1103/htm). Recently, we updated KIPEs with additional features and data sets as described in our preprint ['KIPEs3: Automatic annotation of biosynthesis pathways'](https://doi.org/10.1101/2022.06.30.498365).

<a href="https://github.com/bpucker/KIPEs">
<img alt="KIPEs workflow (Pucker <i>et al.,</i> 2020; doi:10.3390/plants9091103; Rempel&Pucker,2022; doi:10.1101/2022.06.30.498365)" src="https://www.biorxiv.org/content/biorxiv/early/2022/07/02/2022.06.30.498365/F1.large.jpg?width=800&height=600&carousel=1" width="25%" height="25%">
</a>  

### MYB_annotator
This tool performs an automatic identification, annotation, and analysis of the MYB gene family in plants. It can be applied to new transcriptome of genome assemblies. Please find the scripts and data sets in the [MYB_annotator repository](https://github.com/bpucker/MYB_annotator). The MYB_annotator is also available on [our web server](http://pbb.bot.nat.tu-bs.de/MYB_annotator/). This enables all scientists to use KIPEs in their research projects. Please find additional details in the corresponding publication ['Automatic identification and annotation of MYB gene family members in plants'](https://doi.org/10.1186/s12864-022-08452-5).

<a href="https://github.com/bpucker/MYB_annotator">
<img alt="MYB_annotator workflow (Pucker, 2022; doi:10.1186/s12864-022-08452-5)" src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs12864-022-08452-5/MediaObjects/12864_2022_8452_Fig1_HTML.png?as=webp" width="25%" height="25%">
</a>


### Mapping-based Genome Size Estimation (MGSE)
Mapping-based Genome Size Estimation (MGSE) performs an estimation of a genome size based on a read mapping to an existing genome sequence assembly. Please find the script and detailed instructions in the [MGSE repository](https://github.com/bpucker/MGSE). Additional details are described in the corresponding publication ['Mapping-based genome size estimation'](https://doi.org/10.1101/607390).

<a href="https://github.com/bpucker/MGSE">
<img alt="MGSE concept (Pucker, 2021; doi:10.1101/607390)" src="https://raw.githubusercontent.com/bpucker/MGSE/df459458173ec0bcffba7ce61484e5765118c817/MGSE_concept.png" width="50%" height="50%">
</a>


### Neighborhood-Aware Variant Impact Predictor (NAVIP)
Prediction of the functional consequences of sequence variants listed in a VCF file. Please find scripts and detailed instructions in the [NAVIP repository](https://github.com/bpucker/NAVIP). We are currently working to make NAVIP available through our web server. Please find additional details in the corresponding publication ['Influence of neighboring small sequence variants on functional impact prediction'](https://github.com/bpucker/NAVIP).

<a href="https://github.com/bpucker/NAVIP">
<img alt="cInDels detected by NAVIP (Baasner <i>et al</i>., 2019; doi:10.1101/596718 )" src="https://www.biorxiv.org/content/biorxiv/early/2019/06/27/596718/F3.large.jpg?width=800&height=600&carousel=1" width="50%" height="50%">
</a>

## Teaching
### Introduction to Biochemistry and Bioinformatics of Plants
Material used in this course can be found [here](https://github.com/bpucker/teaching/tree/master/MB09_PlantBiochemistryBioinformatics).

<a href="https://github.com/bpucker/teaching/tree/master/MB09_PlantBiochemistryBioinformatics">
<img alt="Digitalis flowers" src="https://github.com/bpucker/teaching/blob/master/MB09_PlantBiochemistryBioinformatics/digitalis_pictures.png?raw=true" width="20%" height="20%">
</a>


### Applied Plant Genomics (in preparation)
Material used in this course can be found [here](https://github.com/bpucker/teaching/tree/master/GE31_AppliedPlantGenomics). Scripts and instructions of the bioinformatics component are available [here](https://github.com/bpucker/AppliedPlantGenomics).

<a href="https://github.com/bpucker/teaching/tree/master/GE31_AppliedPlantGenomics">
<img alt="ONT sequencing" src="https://www.tu-braunschweig.de/index.php?eID=dumpFile&t=p&p=625087&token=b5c0ad2fa83053e994c1789a280edb16e3078b1c" width="20%" height="20%">
</a>


### Data Literacy in Plant Sciences (in preparation) 
Material used in this course can be found [here](https://github.com/bpucker/teaching/tree/master/GE32_DataLiteracyInPlantSciences). Scripts and instructions of the bioinformatics component are available [here](https://github.com/bpucker/DataLiteracyInPlantSciences).


## Project Specific Repositories



## References



## Contact
Do you have questions or are you interested in a collaboration? Please get in touch via [email](https://www.tu-braunschweig.de/en/ifp/pbb/contact).
