# Github Repository Overview
This is an overview of all github repositories associated with [PuckerLab](https://www.pbb.uni-bonn.de). Please find a list of the science communication projects, the developed bioinformatics tools, teaching activities, and project specific repositories below.


## (1) Science Communication
### (1.1) [MybMonday](https://github.com/bpucker/MybMonday)
Specific facts and research projects about MYB transcription factors are shared on a weekly basis via social media using [#MybMonday](https://github.com/bpucker/MybMonday).

<a href="https://github.com/bpucker/MybMonday">
<img alt="Phylogenetic tree of banana MYBs (Pucker <i>et al.</i>, 2020; doi:10.1371/journal.pone.0239275)" src="https://journals.plos.org/plosone/article/figure/image?size=large&id=10.1371/journal.pone.0239275.g002" width="400">
</a>

### (1.2) [FlavonoidFriday](https://github.com/bpucker/FlavonoidFriday)
Specific facts and research projects about the flavonoid biosynthesis are shared on a weekly basis via social media using [#FlavonoidFriday](https://github.com/bpucker/FlavonoidFriday).

<a href="https://github.com/bpucker/FlavonoidFriday">
<img alt="Flavonoid biosynthesis pathway overview (Pucker <i>et al.,</i> 2020; doi:10.3390/plants9091103" src="https://camo.githubusercontent.com/a4ae5610c93b6a51b008a9eee2e826ab1b092e056fb6d1ec2d952dd8d7bd4650/68747470733a2f2f7062732e7477696d672e636f6d2f6d656469612f457534794f4366575941494d4b784a3f666f726d61743d6a7067266e616d653d343039367834303936" width="400">
</a>  


## (2) Tools
### (2.1) Knowledge-based Identification of Pathway Enzymes (KIPEs)
Knowledge-based Identification of Pathway Enzymes (KIPEs) performs an automatic annotation of the flavonoid biosynthesis steps in a new transcriptome of genome sequence assembly. Scripts and datasets are availabe in the [KIPEs repository](https://github.com/bpucker/KIPEs). KIPEs is also available on [our web server](https://pbb-tools.de/KIPEs/). This enables all scientists to use KIPEs in their research projects. Please find additional details in the corresponding publication ['Automatic Identification of Players in the Flavonoid Biosynthesis with Application on the Biomedicinal Plant <i>Croton tiglium</i>'](https://www.mdpi.com/2223-7747/9/9/1103/htm). Recently, we updated KIPEs with additional features and data sets as described in our preprint ['KIPEs3: Automatic annotation of biosynthesis pathways'](https://doi.org/10.1371/journal.pone.0294342).

<a href="https://github.com/bpucker/KIPEs">
<img alt="KIPEs workflow (Pucker <i>et al.,</i> 2020; doi:10.3390/plants9091103; Rempel et al.,2023; doi:10.1371/journal.pone.0294342)" src="https://journals.plos.org/plosone/article/figure/image?size=large&id=10.1371/journal.pone.0294342.g001" width="400">
</a>  

### (2.2) MYB_annotator
This tool performs an automatic identification, annotation, and analysis of the MYB gene family in plants. It can be applied to new transcriptome of genome assemblies. Please find the scripts and data sets in the [MYB_annotator repository](https://github.com/bpucker/MYB_annotator). The MYB_annotator is also available on [our web server](https://pbb-tools.de/MYB_annotator/). This enables all scientists to use the bHLH_annotator in their research projects. Please find additional details in the corresponding publication ['Automatic identification and annotation of MYB gene family members in plants'](https://doi.org/10.1186/s12864-022-08452-5).

<a href="https://github.com/bpucker/MYB_annotator">
<img alt="MYB_annotator workflow (Pucker, 2022; doi:10.1186/s12864-022-08452-5)" src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs12864-022-08452-5/MediaObjects/12864_2022_8452_Fig1_HTML.png?as=webp" width="400">
</a>


### (2.3) Mapping-based Genome Size Estimation (MGSE)
Mapping-based Genome Size Estimation (MGSE) performs an estimation of a genome size based on a read mapping to an existing genome sequence assembly. Please find the script and detailed instructions in the [MGSE repository](https://github.com/bpucker/MGSE). Additional details are described in the corresponding publication ['Mapping-based genome size estimation'](https://doi.org/10.1186/s12864-025-11640-8).

<a href="https://github.com/bpucker/MGSE">
<img alt="MGSE concept (Natarajan, 2025; doi:10.1186/s12864-025-11640-8)" src="https://raw.githubusercontent.com/bpucker/MGSE/df459458173ec0bcffba7ce61484e5765118c817/MGSE_concept.png" width="400">
</a>


### (2.4) Neighborhood-Aware Variant Impact Predictor (NAVIP)
Prediction of the functional consequences of sequence variants listed in a VCF file. Please find scripts and detailed instructions in the [NAVIP repository](https://github.com/bpucker/NAVIP). We are currently working to make NAVIP available through our web server. Please find additional details in the corresponding publication ['NAVIP: Unraveling the influence of neighboring small sequence variants on functional impact prediction'](https://doi.org/10.1371/journal.pcbi.1012732).

<a href="https://github.com/bpucker/NAVIP">
<img alt="cInDels detected by NAVIP (Baasner <i>et al</i>., 2025; doi: 10.1371/journal.pcbi.1012732)" src="https://journals.plos.org/ploscompbiol/article/figure/image?size=large&id=10.1371/journal.pcbi.1012732.g002" width="400">
</a>

### (2.5) bHLH_annotator
This tool performs an automatic identification, annotation, and analysis of the bHLH transcription factor gene family in plants. It can be applied to new transcriptome assemblies or to the annotation derived from genome sequences. Please find the scripts and data sets in the [bHLH_annotator repository](https://github.com/bpucker/bHLH_annotator). The bHLH_annotator is also available on [our web server](https://pbb-tools.de/bHLH_annotator/). This enables all scientists to use the bHLH_annotator in their research projects. Please find additional details in the corresponding publication ['Automatic annotation of the bHLH gene family in plants'](https://doi.org/10.1186/s12864-023-09877-2).

<a href="https://github.com/bpucker/bHLH_annotator">
<img alt="bHLH_annotator workflow (Thoben & Pucker, 2023 doi: 10.1186/s12864-023-09877-2" src="https://github.com/bpucker/bHLH_annotator/blob/main/bHLH_annotator.png?raw=true" width="400">
</a>




## (3) Teaching @ Uni Bielefeld

### (3.1) Applied Python Programming for Life Scientists (APPLS)
This course enables life scientists to learn Python basics to address their specific research questions. The [APPLS repository](https://github.com/bpucker/APPLS) contains the materials of this course.

<a href="https://github.com/bpucker/APPLS">
<img alt="Applied Python Programming for Life Scientists (APPLS)" src="https://github.com/bpucker/figures/raw/main/Python_course.png" width="400">
</a>

### (3.2) Applied Plant Genomics
This course provides a comprehensive introduction to the analysis of next generation sequencing (NGS) data sets. All materials are available in the [AGR repository](https://github.com/bpucker/AppliedGenomeResearch).

<a href="https://github.com/bpucker/AppliedGenomeResearch">
<img alt="Applied Plant Genomics" src="https://github.com/bpucker/figures/raw/main/AppliedGenomeReserachCourseContentOverview.png" width="400">
</a>

### (3.3) Molecular Methods in Genome Research
This course introduces molecular biologists to wet lab methods of genome research. Please find the materials in the [MMGR repository](https://github.com/bpucker/MolecularMethodsInGenomeResearch).

<a href="https://github.com/bpucker/MolecularMethodsInGenomeResearch">
<img alt="Molecular Methods in Genome Research" src="https://github.com/bpucker/figures/raw/main/MolecularMethodsInGenomeReserachCourseOverview.png" width="400">
</a>

## (4) Teaching @ TU Braunschweig
### (4.1) Introduction to Biochemistry and Bioinformatics of Plants (MB09)
Material used in this course can be found [here](https://github.com/bpucker/teaching/tree/master/MB09_PlantBiochemistryBioinformatics).

<a href="https://github.com/bpucker/teaching/tree/master/MB09_PlantBiochemistryBioinformatics">
<img alt="Digitalis flowers" src="https://github.com/bpucker/teaching/blob/master/MB09_PlantBiochemistryBioinformatics/digitalis_pictures.png?raw=true" width="400">
</a>


### (4.2) Applied Plant Genomics (GE31)
Material used in this course can be found [here](https://github.com/bpucker/teaching/tree/master/GE31_AppliedPlantGenomics). Scripts and instructions of the bioinformatics component are available [here](https://github.com/bpucker/AppliedPlantGenomics).

<a href="https://github.com/bpucker/teaching/tree/master/GE31_AppliedPlantGenomics">
<img alt="ONT sequencing" src="https://github.com/bpucker/figures/blob/main/ONT_sequencing.png?raw=true" width="400">
</a>


### (4.3) Data Literacy in Plant Sciences (GE32)
Material used in this course can be found [here](https://github.com/bpucker/teaching/tree/master/GE32_DataLiteracyInPlantSciences). Scripts and instructions of the bioinformatics component are available [here](https://github.com/bpucker/DataLiteracyInPlantSciences).


### (4.4) iGEM @ TU Braunschweig
iGEM (international Genetically Engineered Machine) is the largest synthetic biology competition with over 300 international teams participating each year. [PuckerLab](https://www.pbb.uni-bonn.de) hosted the [TU Braunschweig iGEM2022](https://2022.igem.wiki/tu-braunschweig/) and [TU Braunschweig iGEM2023](https://2023.igem.wiki/tu-braunschweig/) teams.

<a href="https://www.izmb.uni-bonn.de/en/molecular-plant-sciences/igem">
<div>
<img alt="iGEM" src="https://github.com/bpucker/figures/blob/main/igem_logo.png" width="400">
</div>
</a>  


### (4.5) Data Literacy in Genome Research / Data Literacy in Genomics
Material used in this course can be found [here](https://github.com/bpucker/teaching/tree/master/FRX_DataLiteracyInGenomeResearch).


### (4.6) Applied Python Programming for Life Scientists (BB34)
Material used in this course can be found [here](https://github.com/bpucker/PythonTUBS).


### (4.7) Applied Plant Transcriptomics (GE33)
Material used in this course can be found [here](https://github.com/bpucker/AppliedPlantTranscriptomics).


### (4.8) Molecular Plant Biology for Food Chemists (LMChemBSc-12)
Material used in this course can be found [here](https://github.com/bpucker/teaching/tree/master/LMChemBSc-12).


## (5) Teaching @ Uni Bonn
### (5.1) Python for Life Scientists (PFLS)
Material used in this course can be found [here](https://github.com/bpucker/PyBo).

<a href="[https://www.izmb.uni-bonn.de/en/molecular-plant-sciences/igem](https://github.com/bpucker/PyBo)">
<div>
<img alt="Python" src="https://github.com/bpucker/figures/blob/main/Python_course.png?raw=true" width="400">
</div>
</a>  


### (5.2) Introduction to Molecular Plant Biology (BIO13)



Plant Biochemistry, Physiology and Molecular Biology (PBPM) lecture
https://github.com/bpucker/teaching/tree/master/PBPM


Soft skill seminars offered to PuckerLab members
https://github.com/bpucker/teaching/tree/master/SoftSkillSeminars

Long Read Genomics (Lecture series in Prague)
https://github.com/bpucker/LRG

de.NBI courses
https://github.com/bpucker/teaching/tree/master/deNBI2025





## (6) Project Specific Repositories

### (6.1) Evolution of _FNS I_ in the _Apiaceae_ ###
Scripts associated with our analysis of the _FNS I_ evolution in the Apiaceae are available in the [ApiaceaeFNS1](https://github.com/bpucker/ApiaceaeFNS1) repository. Please find additional details in our preprint [_Apiaceae FNS I_ originated from _F3H_ through tandem gene duplication](https://doi.org/10.1101/2022.02.16.480750).

### (6.2) Co-expression analysis ###
The [CoExp](https://github.com/bpucker/CoExp) repository contains several scripts associated with a co-expression analysis. This includes the preparation of count tables with kallisto. The actual co-expression analysis is also available through our [webserver](http://pbb.bot.nat.tu-bs.de/CoExp/).

### (6.3) Expression analysis in pitaya ###
The scripts in the [pitaya](https://github.com/bpucker/pitaya) repository were developed and applied to refute two publications about pigments in this Caryophyllales plant species. Please find additional details in our publication [The report of anthocyanins in the betalain-pigmented genus _Hylocereus_ is not well evidenced and is not a strong basis to refute the mutual exclusion paradigm](https://doi.org/10.1186/s12870-021-03080-9) and preprint [The evidence for anthocyanins in the betalain-pigmented genus _Hylocereus_ is weak](https://doi.org/10.1101/2021.11.16.468878).

### (6.4) Long Read Walker ###
The [LongReadWalker](https://github.com/bpucker/LongReadWalker) allows to close gaps between contigs of long read assemblies. All reads are screened for overlaps with the flanking contigs. Matches are used to extend contigs. An iterative approach is similar to the walking strategy deployed in the early days of genome sequencing when primers were placed at the border of known sequences to 'walk' into unknown sequences. The LongReadWalker tries to close a gap starting at both flanking contigs. The resulting sequences should match to fully support the gap replacing sequence.

### (6.5) Genotyping GABI-Kat lines with ONT long reads ###
The [GKseq](https://github.com/bpucker/GKseq) repository contains some scripts used for the analysis described in our paper [Large scale genomic rearrangements in selected Arabidopsis thaliana T-DNA lines are caused by T-DNA insertion mutagenesis](https://doi.org/10.1186/s12864-021-07877-8). Please find [loreta](https://github.com/nkleinbo/loreta) in this separate repository.

### (6.6) NAVIP ###
Please find [NAVIP](https://github.com/bpucker/NAVIP) in the corresponding repository. Detailed explanations can be found in our preprint [Influence of neighboring small sequence variants on functional impact prediction](https://doi.org/10.1101/596718).

### (6.7) Mapper and variant caller comparison ###
Please find scripts associated with our benchmarking study [Comparison of Read Mapping and Variant Calling Tools for the Analysis of Plant NGS Data](https://www.mdpi.com/2223-7747/9/4/439/htm) in the [variant calling](https://github.com/bpucker/variant_calling) repository.

### (6.8) Yam genome sequence ###
The [yam](https://github.com/bpucker/yam) repository contains several analysis scripts associated with the investigation of the yam genome sequence [High Contiguity _de novo_ Genome Sequence Assembly of Trifoliate Yam (_Dioscorea dumetorum_) Using Long Read Sequencing](https://www.mdpi.com/2073-4425/11/3/274/htm).

### (6.9) Non-canonical splice sites ###
The [ncss](https://github.com/bpucker/ncss2018) repository contains scripts required for the systematic analysis of non-canonical splice sites. Please find additional details in the corresponding publications [Genome-wide analyses supported by RNA-Seq reveal non-canonical splice sites in plant genomes](https://doi.org/10.1186/s12864-018-5360-z) and [Animal, Fungi, and Plant Genome Sequences Harbor Different Non-Canonical Splice Sites](https://www.mdpi.com/2073-4409/9/2/458/htm).

### (6.10) Investigation of the At7 genome ###
The [At7](https://github.com/bpucker/At7) repository contains several scripts associated with the coverage analyses performed on the At7 genome sequencing data sets. Please find additional details in the corresponding publicaton [Twenty-Five Years of Propagation in Suspension Cell Culture Results in Substantial Alterations of the Arabidopsis Thaliana Genome](https://www.mdpi.com/2073-4425/10/9/671/htm).

### (6.11) Nd-1 genome ###
The [Nd-1](https://github.com/bpucker/Nd1_PacBio) repository contains scripts developed for the PacBio sequencing of the _Arabidopsis thaliana_ Nd-1 genome sequence. Please find additional details in the corresponding publication [A chromosome-level sequence assembly reveals the structure of the _Arabidopsis thaliana_ Nd-1 genome and its gene set](https://doi.org/10.1371/journal.pone.0216233).

### (6.12) Analysis of PEPD structure ###
The [PEPD](https://github.com/bpucker/PEPD) repository contains scripts and instructions associated with the analysis of the PEPD structure. Please find additional details about the analysis results in our preprint [Harnessing natural diversity to identify key amino acid residues in prolidase](https://doi.org/10.1101/423475).


https://github.com/bpucker/CaryoAnthoBlock (under review)


## (7) Collection of bioinformatic tools
All tools developed by the [PuckerLab](https://www.pbb.uni-bonn.de) (Uni Bonn) are available in the [PBBtools repository](https://github.com/bpucker/PBBtools). Many of these tools are available via our [BioinfToolServer](https://www.pbb-tools.de).


## References
Please find a full list of our publications on the [@PuckerLab website](https://www.izmb.uni-bonn.de/en/molecular-plant-sciences/publications).


## Contact
Do you have questions or are you interested in a collaboration? Please get in touch via [email](https://www.izmb.uni-bonn.de/en/molecular-plant-sciences/contact).
